mp/src/game/server/momentum/mom_generic_bomb.cpp

```diff
@@ -0,0 +1,118 @@
+﻿//========= Copyright Valve Corporation, All rights reserved. ============//
+//
+// Purpose: Exploding bomb
+//
+//=============================================================================//
+
+#include "cbase.h"
+#include "mom_generic_bomb.h"
+#include "baseanimating.h"
+#include "mom_gamerules.h"
+#include "world.h"
+
+// memdbgon must be the last include file in a .cpp file!!!
+#include "tier0/memdbgon.h"
+
+BEGIN_DATADESC(CMomGenericBomb)
+DEFINE_KEYFIELD(m_flDamage, FIELD_FLOAT, "damage"), DEFINE_KEYFIELD(m_flRadius, FIELD_FLOAT, "radius"),
+    DEFINE_KEYFIELD(m_iHealth, FIELD_INTEGER, "health"),
+    DEFINE_KEYFIELD(m_strExplodeSoundName, FIELD_SOUNDNAME, "sound"),
+
+    DEFINE_OUTPUT(m_Detonate, "OnDetonate"),
+
+    DEFINE_INPUTFUNC(FIELD_VOID, "Detonate", InputDetonate),
+    END_DATADESC()
+
+        PRECACHE_REGISTER(momentum_generic_bomb);
+
+LINK_ENTITY_TO_CLASS(momentum_generic_bomb, CMomGenericBomb);
+
+void CMomGenericBomb::Precache()
+{
+    PrecacheModel(STRING(GetModelName()));
+    PrecacheScriptSound(STRING(m_strExplodeSoundName));
+    BaseClass::Precache();
+}
+
+CMomGenericBomb::CMomGenericBomb()
+{
+    SetMaxHealth(1);
+    SetHealth(1);
+    m_flDamage = 50.0f;
+    m_flRadius = 100.0f;
+    m_takedamage = DAMAGE_YES;
+}
+
+void CMomGenericBomb::Spawn()
+{
+    m_takedamage = DAMAGE_YES;
+
+    char *szModel = (char *)STRING(GetModelName());
+
+    if (!szModel || !*szModel)
+    {
+        Warning("momentum_generic_bomb at %.0f %.0f %0.f missing modelname\n", GetAbsOrigin().x, GetAbsOrigin().y,
+                GetAbsOrigin().z);
```

Comment:
Call `GetAbsOrigin()` once and set it to a variable, use the variable in this print